README.md

```diff
@@ -1,5 +1,49 @@
 # bundle loader for webpack
 
+## What's This?
+
+This is a loader for webpack that provides a way to loader modules async.
+
+Yes, you can do async loading with `require.ensure` or AMD `require` function like this:
+
+```javascript
+require.ensure(['./a','./b'],function(require){/* do something with a and b */})
+```
+
+Webpack will package modules `a` and `b` into the same file, And you can not load them seperately.
+
+When working with dynamic modules, things get worse:
+
+```javascript
+function loadPage(pageName, callback) {
+  try {
+    require(["./pages/" + pageName], function(page) {
+      callback(null, page);
+    });
+  } catch(e) {
+    calback(e);
+  }
+}
+```
+
+All modules in `./pages/` folder will be packaged into one single file! If you want to load a seperate module when in need, you need bundle-loader!
+
+```javascript
+function loadPage(pageName, callback) {
+  try {
+    var pageBundle = require("bundle!./pages/" + pageName)
+  } catch(e) {
+    return callback(e);
```

Comment:
Neither `bundle-loader` nor `require.ensure` throws an error on script loading error. Even more, it won't be caught here since loading is async.